### PR TITLE
[Snapshot] Modify the prefer snapshot version

### PR DESCRIPTION
**Pull request by morningman**
In this CL, prefer snapshot version in snapshot request is defined
in thrift. So that both FE and BE can use this version value.

ISSUE #2432

### CODE REVIEW

File: gensrc/thrift/AgentService.thrift [hunk 1]
```diff
@@ -178,7 +178,7 @@ struct TSnapshotRequest {
     7: optional bool list_files
     // if all nodes has been upgraded, it can be removed.
     8: optional bool allow_incremental_clone
-    9: optional i32 preferred_snapshot_version = 1  // request preferred snapshot version, default value is 1 for old version be
+    9: optional i32 preferred_snapshot_version = Types.TPREFER_SNAPSHOT_REQ_VERSION
```
Comment (kangkaisen) [inline gensrc/thrift/AgentService.thrift]: You change the default value, is there no compatibility ?

Comment (morningman) [reply gensrc/thrift/AgentService.thrift]: No problem, the version 1 is no longer used, and if we set it to 1, snapshot will failed.
Actually, preferred_snapshot_version is a `required` field, and should always be set to
`TPREFER_SNAPSHOT_REQ_VERSION`. Make it as `optional` is just for compatibility.

Comment (kangkaisen) [reply gensrc/thrift/AgentService.thrift]: OK.

**Review by kangkaisen — DISMISSED** [review]
+1. LGTM

File: gensrc/thrift/Types.thrift [hunk 2]
```diff
@@ -380,3 +380,7 @@ struct TUserIdentity {
     3: optional bool is_domain
 }
 
+const i32 TSNAPSHOT_REQ_VERSION1 = 3;
+const i32 TSNAPSHOT_REQ_VERSION2 = 4;
```
Comment (kangpinghuang) [inline gensrc/thrift/Types.thrift]: add some log to describe the major version function.

**Review by kangpinghuang — CHANGES_REQUESTED** [review]

File: gensrc/thrift/Types.thrift (hunk 2 above)
Comment (morningman) [reply gensrc/thrift/Types.thrift]: OK

**Review by kangpinghuang — APPROVED** [review]

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**